WMFComponents/Sources/WMFComponents/Components/Activity/WMFActivityViewModel.swift

```diff
@@ -0,0 +1,210 @@
+        case .read(let count):
+            return localizedStrings.getActivityTabReadTitle(count)
+        case .save(let count):
+            return localizedStrings.getActivityTabSaveTitle(count)
+        case .noEdit:
+            if shouldShowAddAnImage {
+                return localizedStrings.activityTabNoEditsAddImagesTitle
+            } else {
+                return localizedStrings.activityTabNoEditsGenericTitle
+            }
+        }
+    }
+    
+    func action(for type: ActivityTabDisplayType) -> (() -> Void)? {
+        switch type {
+        case .edit:
+            return nil
+        case .read:
+            return openHistory
+        case .save:
+            return openSavedArticles
+        case .noEdit:
+            if shouldShowAddAnImage {
+                return openSuggestedEdits
+            } else {
+                return openStartEditing
+            }
+        }
+    }
+    
+    func backgroundColor(for type: ActivityTabDisplayType) -> UIColor {
+        switch type {
+        case .edit, .noEdit:
+            theme.softEditorBlue
+        case .save:
+            theme.softEditorGreen
+        case .read:
+            theme.softEditorOrange
+        }
+    }
+    
+    func leadingIconColor(for type: ActivityTabDisplayType) -> UIColor {
+        switch type {
+        case .edit, .noEdit:
+            theme.editorBlue
+        case .save:
+            theme.editorGreen
+        case .read:
+            theme.editorOrange
+        }
+    }
+    
+    func trailingIconName(for type: ActivityTabDisplayType) -> String {
+        
+        switch type {
+        case .noEdit:
+            if shouldShowAddAnImage {
+                return "add-images"
+            }
+            return "activity-link"
+        default:
+            return "chevron.forward"
+        }
+    }
+    
+    func borderColor(for type: ActivityTabDisplayType) -> UIColor {
+        switch type {
+        case .edit, .noEdit:
+            WMFColor.blue100
+        case .save:
+            WMFColor.green100
+        case .read:
+            WMFColor.beige100
+        }
+    }
+    
+    func titleFont(for type: ActivityTabDisplayType) -> UIFont {
+        switch type {
+        case .noEdit:
+            return WMFFont.for(.headline)
+        default:
+            return WMFFont.for(.boldHeadline)
+        }
+    }
+
+    func getGroupAssigment() -> WMFActivityTabExperimentsDataController.ActivityTabExperimentAssignment {
+        guard let dataController = WMFActivityTabExperimentsDataController.shared else {
+            return .control
+        }
+        var assignment: WMFActivityTabExperimentsDataController.ActivityTabExperimentAssignment = .control
+
+        do {
+            let currentAssigment = try dataController.getActivityTabExperimentAssignment()
+            assignment = currentAssigment
+        } catch {
+            debugPrint("Error assigning activity tab experiment: \(error)")
```

Comment:
Minor quibble - I don't think this code is actually assigning the experiment, just pulling the existing assignment. Might prevent confusion if we reword: `Error fetching activity tab experiment assignment`.